index.html

```diff
@@ -1415,6 +1415,51 @@ <h4>
       ]});
     </pre>
   </section>
+
+  <section> <h3>Push and read unknown records inside an external record</h3>
+    <p>
+      Unknown type records may be useful inside external type records as
+      developers know what they represent and therefore can avoid specifying the
+      mime type.
+    </p>
+    <pre class="example">
+      const encoder = new TextEncoder();
+      const writer = new NDEFWriter();
+      writer.push({ records: [
+        {
+          recordType: "example.com:shoppingItem", // External record
+          data: {
+            records: [
```

Comment:
Not sure if the order of records is preserved. The app then would not be 100% if the first unknown record is the one they assume to be. I suggest using only one `unknown` record in the message, or include an id. (not that I would endorse using `unknown` records, as `external` is probably better for all use cases. I think `unknown` is mostly as a backup for reading use cases (if nothing works). But I need to double check these.

Reply:
According to the spec below, it looks like it depends on the platform.

> Actual NDEF records do not carry an index number; the ordering is implicitly given by the order in which the NDEF records are serialized. For example, if NDEF records are repackaged by an intermediate application, then that application is responsible for ensuring that the order of NDEF records is preserved.

Reply:
Then having two `unknown` records in a message is a hazard :).

Reply:
But when written as a sub record with web nfc we could enforce order... I dont actually see why changing the order would make sense

Reply:
What I meant was that in an (educative) example it would be better either to use `unknown` + id, or not use two `unknown` records in the same message, but the example intro ("developers know what they represent and therefore can avoid specifying the mime type") makes clear the intention. I am not blocking this, just made a remark.

Reply:
Does that look good to you as well @kenchris?